timeline/static/js/src/timeline_edit.js

```diff
@@ -74,7 +91,37 @@ function TimelineXBlockEdit(runtime, element) {
     }
 
     function collectTimelineData() {
-        return timelineItems;
+        return timelineItems.filter(item => item.content && item.start);
+    }
+
+    function showTab(index) {
+        tabs.forEach((tab, idx) => {
+            tab.toggleClass('hidden', idx !== index);
+        });
+        $(element).find('.back-button').toggleClass('hidden', index === 0);
+        $(element).find('.continue-button').toggleClass('hidden', index === tabs.length - 1);
+        $(element).find('.save-button').parent().toggleClass('hidden', index !== tabs.length - 1);
+    }
+
+    function saveTimelineItemData() {
+        const errorMessageContainer = $(element).find('.error-message');
+        errorMessageContainer.hide().text('');
+
+        if (currentEditIndex >= 0) {
+            const title = $(element).find('#timeline-item-content').val();
+            const date = $(element).find('#timeline-item-date').val();
+
+            if (title && date) {
+                timelineItems[currentEditIndex].start = date;
+                timelineItems[currentEditIndex].content = title;
+                timelineItems[currentEditIndex].description = tinymce.get('timeline-item-description').getContent();
+                timelineItems[currentEditIndex].milestone = $(element).find('.timeline-item-milestone').prop('checked');
+                renderTimelineItemList();
```

Comment:
`renderTimelineItemList` has the following code:
```
 if (timelineItems.length > 0) {
            listContainer.find('li:first').addClass('selected');
            renderTimelineItemDetails(0);
        }
```
This means that after each save, the first element is selected. This happens for instance as soon as the user stops typing when editing the title, i.e. the selected element is changed while the user is updating it.

Reply:
Thanks for catching that @CefBoud. I've pushed a fix. Please take another look.